### PR TITLE
Sed 1402 fe   dragging of the ranger selection does not stop it exactly in the dropped position

### PR DESCRIPTION
**Pull request by lucianbaciu**
This branch includes also:

https://exense.atlassian.net/browse/SED-1402
https://exense.atlassian.net/browse/SED-1485

Because they were strongly related and they were 'semi-fixed' by this.

### CODE REVIEW

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts [hunk 1]
```diff
@@ -161,14 +161,18 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
...
         this.keywords = keywords;
       });
-    this.executionContext
+    this.executionContext.timeSelectionState
       .onActiveSelectionChange()
       .pipe(takeUntil(this.subscriptionsTerminator$))
       .subscribe((newRange) => {
         this.timeSelection = newRange;
-        this.updateTable();
+        this.updateTable().subscribe();
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:169]: Usually, when subscribe invocation appears inside another subscribe - it's a sign that observable chain should be reorganized. For example this one can be implemented as:
```
this.executionContext.timeSelectionState
     .onActiveSelectionChange()
     .pipe(
         tap((newRange) => this.timeSelection = newRange),
         switchMap(() => this.update()),
         takeUntil(this.subscriptionsTerminator$)
     )
     .subscribe();
```

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts [hunk 2]
```diff
@@ -161,14 +161,18 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
...
         // the event is handled from the parent, so no action needed here.
       });
+    this.executionContext.timeSelectionState
+      .onZoomReset()
+      .pipe(takeUntil(this.subscriptionsTerminator$))
+      .subscribe(() => this.handleZoomReset());
 
     this.executionContext
```
Comment (dvladir) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:177]: Same thing
```
    this.executionContext
      .onFiltersChange()
      .pipe(
            switchMap(() => this.updateAllCharts())
            takeUntil(this.subscriptionsTerminator$))
      )
      .subscribe();
```

**Comment by dvladir** [comment]
@lucianbaciu Those remarks about Observable chains - I don't think the currently now it brings issues or affect on performance.
It's more about code styling and thinking in "rxjs" way.
But it's better to avoid of nested subscriptions.
For example if `update` or `updateTable` method would return `hot` observable nested subscription would give us a memory leak

**Review by dvladir — DISMISSED** [review]

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts (hunk 2 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:177]: fixed all of them!

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts [hunk 3]
```diff
@@ -269,7 +279,8 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
   }
 
   onZoomReset() {
-    this.timeSelectionComponent.resetZoom();
+    this.executionContext.timeSelectionState.resetZoom();
+    // this.timeSelectionComponent.resetZoom();
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:283]: Can't we remove this comment?

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts [hunk 4]
```diff
@@ -568,13 +579,14 @@ export class PerformanceViewComponent implements OnInit, OnDestroy {
...
-    this.summaryChart.resetZoom();
-    this.throughputChart.resetZoom();
-    this.updateTable();
+  handleZoomReset() {
+    // TODO uncomment these when the uplot sync feature is removed
+    // this.responseTimeChart.resetZoom();
+    // this.byStatusChart.resetZoom();
+    // this.byStatusChart.resetZoom();
+    // this.summaryChart.resetZoom();
+    // this.throughputChart.resetZoom();
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/performance-view.component.ts:588]: Will this all work still without these commented out lines? What is the plan for th euplot sync feature to be removed?

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts [hunk 5]
```diff
@@ -27,11 +27,14 @@ export class TimeseriesTableComponent implements OnInit, OnDestroy {
...
-  @Input() executionContext!: ExecutionContext;
+  @Input() executionContext!: TimeSeriesContext;
 
   @Output() onInitializationComplete = new EventEmitter<void>();
 
-  subscriptionsTerminator$ = new Subject<void>();
+  private terminator$ = new Subject<void>();
+
+  // @Output('onKeywordsFetched') onKeywordsFetched = new EventEmitter<string[]>();
+  // @Output('onKeywordToggled') onKeywordToggled = new EventEmitter<string>();
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts:37]: Why did you add two commented out Outputs? Do we need them?

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts [hunk 6]
```diff
@@ -95,32 +99,30 @@ export class TimeseriesTableComponent implements OnInit, OnDestroy {
...
+        let tableBuckets = response.matrix
+          .map((series, i) => {
+            if (series.length != 1) {
+              // we should have just one bucket
+              throw new Error('Something went wrong');
+            }
+            let seriesAttributes = response.matrixKeys[i];
+            let bucket = series[0];
+            bucket.attributes = seriesAttributes;
+            let seriesKey = this.getSeriesKey(seriesAttributes, groupDimensions);
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts:111]: looks like most if not all "let" in this function should be const.

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts [hunk 7]
```diff
@@ -113,48 +121,37 @@ export class PerformanceViewTimeSelectionComponent implements OnInit, OnDestroy
...
-    this.timeRangePicker?.selectFullRange();
-  }
-
   handleTimePickerSelectionChange(timeSelection: TimeRangePickerSelection) {
     let selectionToEmit: ExecutionTimeSelection = { type: timeSelection.type };
     if (timeSelection.type === RangeSelectionType.FULL) {
-      this.rangerComponent.resetSelect(false);
+      this.timeSelectionState.resetZoom();
+      return;
+      // this.rangerComponent.resetSelect(false);
```
Comment (neogucky) [inline projects/step-frontend/src/lib/modules/timeseries/performance-view/time-selection/performance-view-time-selection.component.ts:129]: Can we remove thos commented out line?

**Review by neogucky — COMMENTED** [review]
Thank you for the great PR @lucianbaciu , I added some comments, mostly regarding commented out code (at some point I gave up but basically it is valid for all commented out code in the PR) and a general question regarding the current plans to remove the uplot sync feature. If that is cleared I will merge.

**Comment by lucianbaciu** [comment]
@neogucky I will definitely take care of the commented lines (shame on me)...
uPlot sync feature is great but it has some limitation and add a bit of boiler plate. I don't plan to to remove it now, but the code is somehow organized for an easy replacement of it - because I wouldn't be surprised if we get there.

File: projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts (hunk 5 above)
Comment (lucianbaciu) [reply projects/step-frontend/src/lib/modules/timeseries/performance-view/table/timeseries-table.component.ts:37]: No. I will clean them up. The communication is done via services now

**Review by neogucky — APPROVED** [review]
